AsyncDisplayKit/ASCollectionView.mm

```diff
@@ -1081,6 +1081,9 @@ - (void)didMoveToWindow
   if (!visible && node.inHierarchy) {
     [node __exitHierarchy];
   }
+    
+  // Trigger updating interfaceState for cells in case ASCollectionView becomes visible or invisible
+  [_rangeController visibleNodeIndexPathsDidChangeWithScrollDirection:self.scrollDirection];
```

Comment:
Because we cannot necessarily rely on the call order for the UIkit method, does this always work for nested range controllers? For example, vertically scrolling elements within a pager, or horizontally scrolling elements within a vertical collection?

I'm thinking it could be possible for one of them to re-run the range calculations before it is marked invisible, or something like that.  Note that I think __exitHierarchy may not actually mark it as not visible if the _hierarchyState includes rangeManaged...I am not 100% sure without looking a bit deeper, just have the sense there could be an issue here.